src/app/static/src/app/store/downloadResults/actions.ts

```diff
@@ -1,61 +1,72 @@
 import {ActionContext, ActionTree} from "vuex";
 import {RootState} from "../../root";
-import {DownloadResultsState, DOWNLOAD_TYPE} from "./downloadResults";
+import {DownloadResultsState} from "./downloadResults";
 import {api} from "../../apiService";
 import {DownloadResultsMutation} from "./mutations";
 import {ModelStatusResponse} from "../../generated";
+import {DOWNLOAD_TYPE} from "../../types";
 
 export interface DownloadResultsActions {
     prepareSummaryReport: (store: ActionContext<DownloadResultsState, RootState>) => void
     prepareSpectrumOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
     prepareCoarseOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
+    prepareOutputs: (store: ActionContext<DownloadResultsState, RootState>) => void
     poll: (store: ActionContext<DownloadResultsState, RootState>, downloadType: string) => void
 }
 
 export const actions: ActionTree<DownloadResultsState, RootState> & DownloadResultsActions = {
 
+    async prepareOutputs(context) {
+        const {dispatch} = context
+        await Promise.all([
+            dispatch("prepareCoarseOutput"),
+            dispatch("prepareSummaryReport"),
+            dispatch("prepareSpectrumOutput")
+        ]);
+    },
+
     async prepareCoarseOutput(context) {
         const {state, dispatch, rootState} = context
-
-        const calibrateId = rootState.modelCalibrate.calibrateId
-
-        const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
-            .withSuccess(DownloadResultsMutation.PreparingCoarseOutput)
-            .withError(DownloadResultsMutation.CoarseOutputError)
-            .get(`download/submit/coarse-output/${calibrateId}`)
-
-        if (response) {
-            await dispatch("poll", DOWNLOAD_TYPE.COARSE)
+        if (!state.coarseOutput.downloadId && !state.coarseOutput.preparing) {
```

Comment:
The idea here is that this can be triggered any time, but if there is already an output prepared or in preparation it will not re-request it. When the prepared outputs are invalidated by re-calibrating the model, we just clear the `downloadId` so that the next call to this method will request a new output as required.